hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/math/MultiplyEndToEndTest.java

```diff
@@ -0,0 +1,475 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.math;
+
+import com.hazelcast.sql.impl.expression.ExpressionEndToEndTestBase;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.BigDecimal;
+
+import static com.hazelcast.sql.SqlColumnType.BIGINT;
+import static com.hazelcast.sql.SqlColumnType.DECIMAL;
+import static com.hazelcast.sql.SqlColumnType.DOUBLE;
+import static com.hazelcast.sql.SqlColumnType.INTEGER;
+import static com.hazelcast.sql.SqlColumnType.REAL;
+import static com.hazelcast.sql.SqlColumnType.SMALLINT;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class MultiplyEndToEndTest extends ExpressionEndToEndTestBase {
+
+    @Test
+    public void testBoolean() {
+        assertParsingError("booleanTrue * booleanTrue", "Cannot apply '*' to arguments of type '<BOOLEAN> * <BOOLEAN>'");
+
+        assertParsingError("booleanTrue * byte1", "Cannot apply '*' to arguments of type '<BOOLEAN> * <TINYINT>'");
+        assertParsingError("booleanTrue * short1", "Cannot apply '*' to arguments of type '<BOOLEAN> * <SMALLINT>'");
+        assertParsingError("booleanTrue * int1", "Cannot apply '*' to arguments of type '<BOOLEAN> * <INTEGER>'");
+        assertParsingError("booleanTrue * long1", "Cannot apply '*' to arguments of type '<BOOLEAN> * <BIGINT>'");
+
+        assertParsingError("booleanTrue * float1", "Cannot apply '*' to arguments of type '<BOOLEAN> * <REAL>'");
+        assertParsingError("booleanTrue * double1", "Cannot apply '*' to arguments of type '<BOOLEAN> * <DOUBLE>'");
+
+        assertParsingError("booleanTrue * decimal1", "Cannot apply '*' to arguments of type '<BOOLEAN> * <DECIMAL(38, 38)>'");
+        assertParsingError("booleanTrue * bigInteger1", "Cannot apply '*' to arguments of type '<BOOLEAN> * <DECIMAL(38, 38)>'");
+
+        assertParsingError("booleanTrue * string1", "Cannot apply '*' to arguments of type '<BOOLEAN> * <BOOLEAN>'");
```

Comment:
Why do we expect here `<BOOLEAN> * <BOOLEAN>` instead of VARCHAR?

Reply:
That's because of implicit coercion of varchar to boolean, we have an issue to improve such messages https://github.com/hazelcast/hazelcast/issues/17301